packages/twenty-front/src/modules/activities/events/components/EventRow.tsx

```diff
@@ -135,47 +144,108 @@ export const EventRow = ({
   const diff: Record<string, { before: any; after: any }> = properties?.diff;
 
   const isEventType = (type: 'created' | 'updated') => {
-    return (
-      event.name === type + '.' + targetableObject.targetObjectNameSingular
-    );
+    if (event.name.includes('.')) {
+      return event.name.split('.')[1] === type;
+    }
+    return false;
   };
 
+  const { getIcon } = useIcons();
+
+  const linkedObjectMetadata = useRecoilValue(
+    objectMetadataItemFamilySelector({
+      objectName: event.linkedObjectMetadataId,
+      objectNameType: 'id',
+    }),
+  );
+
```

Comment:
I think we'll need to think about the right mental model for the different types of timeline items. I kept the logic which was to build it dynamically (but moved it on the TS part / away from the JSX part) but it feels like it might become hard to read if we add more and more conditions